packages/react/src/publication/useHidePublication.ts

```diff
@@ -0,0 +1,36 @@
+import { PublicationOwnedByMeFragment } from '@lens-protocol/api-bindings';
+import { invariant } from '@lens-protocol/shared-kernel';
+import { useState } from 'react';
+
+import { useHidePublicationController } from './adapters/useHidePublicationController';
+
+export type HidePublicationRequest = {
+  publication: PublicationOwnedByMeFragment;
+};
+
+export function useHidePublication() {
+  const [isPending, setIsPending] = useState(false);
+  const { hide: hidePublication } = useHidePublicationController();
+
+  const hide = async ({ publication }: HidePublicationRequest) => {
+    invariant(
+      publication.ownedByMe,
+      'Publication not owned by the active profile. Make sure that publication is owned by the active profile (for .e.g. by calling `isPublicationOwnedByMe`) before trying to hide it?',
+    );
```

Comment:
It doesn't have to be the active profile to be able to to hide it.

I think a more accurate and correct approach would be to to check if the current wallet owns the profile author of the publication: `publication.profile.ownedByMe`

Reply:
Makes sense, fixed